### PR TITLE
Update unk_0204CDDC.c with semantics

### PR DESCRIPTION
**Pull request by patilise**
Pretty much the same logic from pret/pokediamond/blob/master/arm9/src/scrcmd_amity_square.c

### CODE REVIEW

File: src/unk_0204CDDC.c [hunk 1]
```diff
@@ -31,84 +33,84 @@ BOOL sub_0204CDF0 (UnkStruct_0203E724 * param0)
...
-        v0 = 5;
-    } else if (v2 < (15 + 15 + 15 + 15 + 10 + 10 + 8)) {
-        v0 = 6;
-    } else if (v2 < (15 + 15 + 15 + 15 + 10 + 10 + 8 + 5)) {
-        v0 = 7;
-    } else if (v2 < (15 + 15 + 15 + 15 + 10 + 10 + 8 + 5 + 5)) {
-        v0 = 8;
+    u8 i, j;
+    u16 rand;
+    u16 * ret_ptr = inline_0204FCAC(ctx);
```
Comment (luckytyphlosion) [inline src/unk_0204CDDC.c:48]: `ret_ptr` -> `retPtr`

**Review by luckytyphlosion — CHANGES_REQUESTED** [review]

**Comment by lhearachel** [comment]
Is there any extra info on what exactly these functions do? They look to me like they're related to accessories: https://github.com/pret/pokediamond/blob/master/arm9/src/scrcmd_amity_square.c#L10

**Comment by patilise** [comment]
They are functions that determine items/accessories that your walking Pokémon picks up in Amity Square, and that a man in Amity Square gives.
Also refer to: https://bulbapedia.bulbagarden.net/wiki/Amity_Square

**Comment by PikalaxALT** [comment]
Is this PR abandoned?

**Pull request merged**